src/pages/TokenDetails/index.tsx

```diff
@@ -80,8 +81,29 @@ export default function TokenDetails() {
   const token = useToken(tokenAddress)
 
   const tokenWarning = tokenAddress ? checkWarning(tokenAddress) : null
-  /* network balance handling */
+  const [continueSwap, setContinueSwap] = useState<{ resolve: (value: boolean | PromiseLike<boolean>) => void }>()
```

Comment:
nit: Define `[continueSwap, setContinueSwap]` immediately before use. In other words, swap L84 and L86.